DiskSpaceMonitoringFailOverTest.shouldNotBeHealthyAfterLeaderChangeWhenNoDiskSpaceAvailable

**Summary**

- How often does the test fail? At least twice on the same branch (see https://github.com/zeebe-io/zeebe/pull/5367)
- Does it block your work? No
- Do we suspect that it is a real failure? Not really, but it's weird it happened twice when updating a dependency for metrics...

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
[2020-09-17T12:55:30.791Z] [ERROR] io.zeebe.broker.it.health.DiskSpaceMonitoringFailOverTest.shouldNotBeHealthyAfterLeaderChangeWhenNoDiskSpaceAvailable  Time elapsed: 120.465 s  <<< ERROR!

[2020-09-17T12:55:30.792Z] org.junit.runners.model.TestTimedOutException: test timed out after 120 seconds

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/jdk.internal.misc.Unsafe.park(Native Method)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:234)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1798)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3128)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1868)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker.lambda$atomixCreateStep$15(Broker.java:221)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker$$Lambda$239/0x00000008402f1840.close(Unknown Source)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.CloseProcess.lambda$closingStepByStep$0(CloseProcess.java:56)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.CloseProcess$$Lambda$942/0x00000008406c4440.run(Unknown Source)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.CloseProcess.closingStepByStep(CloseProcess.java:56)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.CloseProcess$$Lambda$941/0x00000008406c4040.run(Unknown Source)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.bootstrap.CloseProcess.closeReverse(CloseProcess.java:37)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker.lambda$close$25(Broker.java:416)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker$$Lambda$940/0x00000008406c4c40.accept(Unknown Source)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CompletableFuture.uniAcceptNow(CompletableFuture.java:753)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:731)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2108)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker.lambda$close$26(Broker.java:414)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker$$Lambda$939/0x00000008406c4840.run(Unknown Source)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.util.LogUtil.doWithMDC(LogUtil.java:21)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.Broker.close(Broker.java:409)

[2020-09-17T12:55:30.792Z] 	at app//io.zeebe.broker.it.clustering.ClusteringRule$$Lambda$1034/0x000000084072fc40.accept(Unknown Source)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1675)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinPool$WorkQueue.helpCC(ForkJoinPool.java:1115)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinPool.externalHelpComplete(ForkJoinPool.java:1957)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinTask.tryExternalHelp(ForkJoinTask.java:378)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinTask.externalAwaitDone(ForkJoinTask.java:323)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:412)

[2020-09-17T12:55:30.792Z] 	at java.base@11.0.3/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:661)

[2020-09-17T12:55:30.793Z] 	at app//io.zeebe.broker.it.clustering.ClusteringRule.after(ClusteringRule.java:238)

[2020-09-17T12:55:30.793Z] 	at app//org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:59)

[2020-09-17T12:55:30.793Z] 	at app//org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

[2020-09-17T12:55:30.793Z] 	at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)

[2020-09-17T12:55:30.793Z] 	at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:264)

[2020-09-17T12:55:30.793Z] 	at java.base@11.0.3/java.lang.Thread.run(Thread.java:834)

</pre>
</details>

**Hypotheses**

No idea unfortunately

**Logs**

Unfortunately the file was so large Jenkins failed to parse it.

--- a/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java
+++ b/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java
@@ -630,7 +630,7 @@ public final class ClusteringRule extends ExternalResource {
   public void waitForTopology(final Predicate<List<BrokerInfo>> topologyPredicate) {
     Awaitility.await()
         .pollInterval(Duration.ofMillis(100))
-        .atMost(Duration.ofSeconds(10))
+        .atMost(Duration.ofSeconds(60))
         .ignoreExceptions()
         .until(() -> getTopologyFromClient().getBrokers(), topologyPredicate);
   }

--- a/qa/integration-tests/src/test/java/io/zeebe/broker/it/health/DiskSpaceMonitoringFailOverTest.java
+++ b/qa/integration-tests/src/test/java/io/zeebe/broker/it/health/DiskSpaceMonitoringFailOverTest.java
@@ -71,6 +71,9 @@ public class DiskSpaceMonitoringFailOverTest {
     clusteringRule.stopBrokerAndAwaitNewLeader(leaderId);
     final var newLeaderId = clusteringRule.getLeaderForPartition(1).getNodeId();
 
+    // Force rescan of healthcheck
+    clusteringRule.getClock().addTime(Duration.ofSeconds(60));
+
     // then
     Awaitility.await()
         .timeout(Duration.ofSeconds(60))

